addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1505,34 +1513,120 @@ export class OdooEditor extends EventTarget {
         this._collabSelectionsContainer.append(caretElement, ...indicators);
     }
+    /**
+     * Takes style attribute values for `top` and `left`, and returns a copy of
+     * the `_collabSelectionInfos` map, filtered to only contain clients whose
+     * avatar is at that position.
+     *
+     * @see this._collabSelectionInfos
+     * @param {string} top
+     * @param {string} left
+     * @returns {Map<string, SelectionInfo>}
+     */
+    _getClientsAtPosition(top, left) {
+        return new Map([...this._collabSelectionInfos.entries()].filter(([key, value]) => {
+            const avatarTop = value.avatarElement && value.avatarElement.style.top;
+            const avatarLeft = value.avatarElement && value.avatarElement.style.left;
+            return value.avatarElement && avatarTop === top && avatarLeft === left;
+        }));
+    }
+    /**
+     * Takes a filtered version of the `_collabSelectionInfos` map containing
+     * clients whose avatars are at the same position, and update its avatars'
+     * `title` and `data-overlapping-avatars` attributes.
+     *
+     * @param {Map<string, SelectionInfo>} clients
+     */
+    _updateClientAvatarValues(clients) {
+        const allAvatarsInDom = [...this._collabSelectionsContainer.querySelectorAll('.oe-collaboration-caret-avatar')];
+        const avatars = [...clients.values()].map(client => client.avatarElement);
+        const lastInDom = allAvatarsInDom.find(avatar => avatars.includes(avatar));
+        const newTitle = [...clients.values()].map(client => client.clientName).join('\n');
+        for (const client of clients.values()) {
+            const avatar = client.avatarElement;
+            // Only show the number of overlapping avatars on the z-top element,
+            // and then only if there are indeed more than one at the same
+            // position.
+            if (clients.size > 1 && avatar === lastInDom) {
+                avatar.setAttribute('data-overlapping-avatars', clients.size);
+            } else {
+                avatar.removeAttribute('data-overlapping-avatars');
+            }
+            if (avatar.firstElementChild.getAttribute('title') !== newTitle) {
+                avatar.firstElementChild.setAttribute('title', newTitle);
+            }
+        };
+    }
 
     multiselectionRemove(clientId) {
         this._collabSelectionInfos.delete(clientId);
-        this._multiselectionRemoveClient(clientId);
+        this._multiselectionRemoveClient(clientId, true);
     }
 
-    _multiselectionRemoveClient(clientId) {
+    _multiselectionRemoveClient(clientId, removeAvatars) {
         const elements = this._collabSelectionsContainer.querySelectorAll(
             `[data-selection-client-id="${clientId}"]`,
         );
         for (const element of elements) {
-            element.remove();
+            if (!element.closest('.oe-collaboration-caret-avatar') || removeAvatars) {
+                element.remove();
+            }
```

Comment:
Same here actually. Can't we just target things that are not inside the avatar in the selector so that we do not have to filter them here ? Oh but that's going away anyway if we remove the avatars separately.

Reply:
See above :-)